CSharpSourceCode/HarmonyPatches/CharacterObjectPatches.cs

```diff
@@ -1,25 +1,39 @@
 ﻿using HarmonyLib;
+using System;
 using TaleWorlds.CampaignSystem;
+using TaleWorlds.Library;
 using TOW_Core.Utilities.Extensions;
 
 namespace TOW_Core.HarmonyPatches
 {
-    public class CharacterObjectPatches
+    [HarmonyPatch]
+    public static class CharacterObjectPatches
     {
-        [HarmonyPatch]
-        class Patch
+        [HarmonyPrefix]
+        [HarmonyPatch(typeof(CharacterObject), "TroopWage", MethodType.Getter)]
+        public static bool TroopWagePrefix(ref int __result, CharacterObject __instance)
         {
-            [HarmonyPrefix]
-            [HarmonyPatch(typeof(CharacterObject), "TroopWage", MethodType.Getter)]
-            public static bool TroopWagePrefix(ref int __result, CharacterObject __instance)
+            if (__instance.IsUndead())
             {
-                if (__instance.IsUndead())
-                {
-                    __result = 0;
```

Comment:
Jax and Kriegan were already discussing, technical, vampires are not undead, but some stuff is actually applying for them aswell... when you do this we have to from now on should never  set Vampire troops to undead. Also do you account for Skeleton King?

Reply:
I know nothing about Skeleton King so would be nice to get more info i have to account. I will make this script check for undead or vampire?